sentry-samples/sentry-samples-android/src/main/java/io/sentry/samples/android/MainActivity.java

```diff
@@ -146,6 +148,37 @@ protected void onCreate(Bundle savedInstanceState) {
           }
         });
 
+    binding.openSecondActivity.setOnClickListener(
+        view -> {
+          // finishing so its completely destroyed
+          finish();
+          startActivity(new Intent(this, SecondActivity.class));
+        });
+
     setContentView(binding.getRoot());
   }
+
+  @Override
+  protected void onResume() {
+    super.onResume();
+    System.out.println("do some stuff resume");
+
+    finishTransaction(SpanStatus.OK);
+  }
+
+  @Override
+  protected void onStart() {
+    super.onStart();
+    System.out.println("do some stuff start");
+  }
+
+  private void finishTransaction(SpanStatus status) {
+    Sentry.configureScope(
+        scope -> {
+          final ITransaction transaction = scope.getTransaction();
+          if (transaction != null) {
+            transaction.finish(status);
+          }
+        });
```

Comment:
`h`: I really don't like this API. We must find a better way IMO.

Reply:
What about having something like `Sentry.finishTransaction(this)`? We do the magic internally, by somehow calling the integration and then finish the transaction. Of course, we need to make sure to not create memory leaks.

Reply:
100% agreed and already discussed, see: https://github.com/getsentry/sentry-java/blob/feat/auto-activity-tracing/sentry-samples/sentry-samples-android/src/main/java/io/sentry/samples/android/MainActivity.java#L178

Reply:
@philipphofmann what would be `this`? the `Activity` Context? the way how to access the transaction is thru the Scope, the only thing that comes to my mind is via the Static API like JS/Python does, Sentry.getSpan() but not accessing an integration that has a WeakRef ma, this would also make racey as it could be called from different threads during Screen's lifecycle.
I didn't spend time thinking about this yet, the new API should also be compatible with #1286 which makes a bit harder as parts are moving as we speak

Reply:
this is now obsolete by `Sentry.getSpan().finish(...)`